apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatInfoView.kt

```diff
@@ -220,100 +220,230 @@ sealed class SendReceipts {
   }
 }
 
-fun deleteContactDialog(chat: Chat, chatModel: ChatModel, close: (() -> Unit)? = null) {
-  val chatInfo = chat.chatInfo
-  if (chatInfo is ChatInfo.Direct) {
-    AlertManager.shared.showAlertDialogButtonsColumn(
-      title = generalGetString(MR.strings.delete_contact_question),
-      buttons = {
-        Column {
-          // Delete contact
-          SectionItemView({
-            AlertManager.shared.hideAlert()
-            notifyDeleteContactDialog(chat, chatModel, close, contactDeleteMode = ContactDeleteMode.Full())
-          }) {
-            Text(generalGetString(MR.strings.button_delete_contact), Modifier.fillMaxWidth(), textAlign = TextAlign.Center, color = MaterialTheme.colors.error)
-          }
-          if (!chatInfo.contact.chatDeleted) {
-            // Delete contact, keep conversation
-            SectionItemView({
-              AlertManager.shared.hideAlert()
-              notifyDeleteContactDialog(chat, chatModel, close, contactDeleteMode = ContactDeleteMode.Entity())
-            }) {
-              Text(generalGetString(MR.strings.delete_contact_keep_conversation), Modifier.fillMaxWidth(), textAlign = TextAlign.Center, color = MaterialTheme.colors.error)
-            }
+
+enum class ContactDeletionKind {
```

Comment:
this seems like an unnecessary indirection, we could directly check conditions in deleteContactDialog